asp.in

```diff
@@ -338,8 +338,33 @@ dispatch_action() {
 
 initialize || log_fatal 'failed to initialize asp repository in %s' "$ASPROOT"
 
-while getopts ':a:fhV' flag; do
+while getopts ':a:fhV-:' flag; do
   case $flag in
+    -)
+      case "${OPTARG}" in
+        arch=*)
+          [[ -n "${OPTARG#*=}" ]] && OPT_ARCH="${OPTARG#*=}"
+          ;;
+        arch)
+          if [[ -n "${!OPTIND}" ]] ; then
+            OPT_ARCH="${!OPTIND}"
+            OPTIND=$(( $OPTIND + 1 ))
+          else
+            log_fatal "option '-%s' requires an argument" "$OPTARG"
+          fi
+          ;;
+        force)
+          OPT_FORCE=1
+          ;;
+        help)
+          usage
+          exit 0
+          ;;
+        version)
+          version
+          exit 0
+          ;;
+      esac;;
```

Comment:
newline between esac and ;;

And, what about the default case? We shouldn't just silently ignore `--ponies`